src/examples/input-clearable/input-clearable-example.html

```diff
@@ -0,0 +1,4 @@
+<md-input-container class="example-input-container">
+  <input mdInput type="text" placeholder="Clearable input" [(ngModel)]="value"/>
+  <md-icon mdSuffix *ngIf="value" (click)="value=''">close</md-icon>
```

Comment:
So actually... `md-icon-button` doesn't really work well as a suffix currently. It's too tall and it causes the placeholder and input text to be misaligned. Filed #4382 to fix this